app/controllers/plans_controller.rb

```diff
@@ -1,15 +1,50 @@
 class PlansController < ApplicationController
+  def index
 
-    def create
-      @user = User.find(params[:user_id])
-      @plan = @user.plans.create(plan_params)
+  end
+
+  def new
+     @plan = Plan.new
+  end
+
+  def create
+    @user = User.find(current_user.id)
+    @plan = @user.plans.new(plan_params)
+
+    respond_to do |format|
+      if @plan.save
```

Comment:
Use @user.plans.create(plan_params). No need to use @plan = @user.plans.new(plan_params)